test/logformats_test.py

```diff
@@ -38,20 +38,25 @@
 
 def _test_writer_and_reader(test_case, writer_constructor, reader_constructor, sleep_time=None,
                             check_remote_frames=True, check_error_frames=True,
-                            check_comments=False):
+                            check_comments=False, round_timestamps=False):
     """Tests a pair of writer and reader by writing all data first and
     then reading all data and checking if they could be reconstructed
     correctly.
 
-    :param test_case: the test case the use the assert methods on
-    :param sleep_time: specifies the time to sleep after writing all messages.
+    :param unittest.TestCase test_case: the test case the use the assert methods on
+    :param Callable writer_constructor: the constructor of the writer class
+    :param Callable reader_constructor: the constructor of the reader class
+
+    :param float sleep_time: specifies the time to sleep after writing all messages.
         gets ignored when set to None
-    :param check_remote_frames: if true, also tests remote frames
-    :param check_error_frames: if true, also tests error frames
-    :param check_comments: if true, also inserts comments at some
+    :param bool check_remote_frames: if True, also tests remote frames
+    :param bool check_error_frames: if True, also tests error frames
+    :param bool check_comments: if True, also inserts comments at some
         locations and checks if they are contained anywhere literally
         in the resulting file. The locations as selected randomly
         but deterministically, which makes the test reproducible.
+    :param bool round_timestamps: if True, rounds timestamps using :meth:`~builtin.round`
+        before comparing the read messages/events
```

Comment:
Maybe it's better to specify number of decimal places to match against? ASC files only store 4 decimals currently.

Reply:
Well, I think the problem comes from the timestamps being relative to the start of the measurement, but that start timestamp is being serialized without sub-second precision. One failing test case failed because of something like `1000000.197 != 1000000.2`. So I guess rounding is more appropriate *in the case of ASC files*.

Reply:
@christiansandberg Do you agree with that? Shall we merge it?

Reply:
Right, didn’t think of that. Go ahead and merge.